libminifi/include/core/state/nodes/MetricsBase.h

```diff
@@ -228,6 +228,20 @@ class NodeReporter {
    */
   virtual int16_t getMetricsNodes(std::vector<std::shared_ptr<ResponseNode>> &metric_vector, uint16_t metricsClass) = 0;
 
+  /**
+   * Retrieves agent information with manifest only from this source.
+   * @param manifest_vector -- manifest nodes vector.
+   * @return 0 on Success, -1 on failure
+   */
+  virtual int16_t getManifestNodes(std::vector<std::shared_ptr<state::response::ResponseNode>>& manifest_vector) const = 0;
+
+  /**
+   * Returns a response node containing all agent information with manifest and agent status
+   * @return a shared pointer to agent information
+   */
+  virtual std::shared_ptr<state::response::ResponseNode> getAgentInformation() const {
+      return nullptr;
```

Comment:
This looks a bit dangerous.
I don't really like functions returning nullptr by default.
Either:
-Make it pure virtual
-Return empty node to make sure the caller won't crash at least
-Throw an exception here in case this should never be called

Reply:
Made it pure virtual.